addOns/client/src/main/java/org/zaproxy/addon/client/ExtensionClientIntegration.java

```diff
@@ -863,4 +874,27 @@ public void sessionModeChanged(Mode mode) {
             }
         }
     }
+
+    public ClientMap getClientTree() {
+        return clientTree;
+    }
+
+    public void exportClientMap(String path) {
+        File file = new File(path);
+        Writer fileWriter = null;
+
+        try {
+            fileWriter = new FileWriter(file, false);
+        } catch (IOException e) {
+            LOGGER.warn(
+                    "Failed to create file writer for: {}, while exporting Client Map", path, e);
+        }
+
+        try {
+            ClientMapWriter.exportClientMap(fileWriter, getClientTree());
+        } catch (IOException e) {
+            LOGGER.warn("Failure while exporting Client Map: {}", path, e);
+        }
+        LOGGER.info("Wrote: {}", file.getAbsolutePath());
```

Comment:
The logs need to be tidied up, either failed or wrote, I don't think this worth info.

Reply:
No worries. Yes it's close enough. I need the push moving it along.